code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/dao/impl/AuthDaoImpl.java

```diff
@@ -129,70 +160,221 @@ public void updateAccount(Account account, String name, String kind, String exte
+    public Boolean hasAccessToProject (Long projectId, Long policyAccount, Boolean isAdmin, Set<ExternalId> externalIds) {
+        return isProjectMember(projectId, policyAccount, isAdmin, externalIds);
     }
     
-    public ObjectManager getObjectManager() {
-        return objectManager;
+    public Boolean isProjectOwner (Long projectId, Long policyAccount, Boolean isAdmin, Set<ExternalId> externalIds){
+        if (projectId == null || externalIds == null) {
+            return false;
+        }
+        if (isAdmin) {
+            return true;
+        }
+        if (policyAccount.equals(projectId)){
+            return false;
+        }
+        Set<ProjectMemberRecord> projectMembers = new HashSet<>();
+        Condition allMembers = DSL.falseCondition();
+        for (ExternalId id: externalIds){
+            allMembers = allMembers.or(PROJECT_MEMBER.EXTERNAL_ID.eq(id.getId())
+                    .and(PROJECT_MEMBER.EXTERNAL_ID_TYPE.eq(id.getType()))
+                    .and(PROJECT_MEMBER.ROLE.eq(ProjectConstants.OWNER))
+                    .and(PROJECT_MEMBER.PROJECT_ID.eq(projectId))
+                    .and(PROJECT_MEMBER.STATE.eq(CommonStatesConstants.ACTIVE))
+                    .and(PROJECT_MEMBER.REMOVED.isNull()));
+        }
+        projectMembers.addAll(create().selectFrom(PROJECT_MEMBER).where(allMembers).fetch());
+        return !projectMembers.isEmpty();
+    }
+    
+    public Boolean isProjectMember (Long projectId, Long policyAccount, Boolean isAdmin,Set<ExternalId> externalIds){
+        if (projectId == null || externalIds == null) {
+            return false;
+        }
+        if ((policyAccount != null && policyAccount.equals(projectId)) || isAdmin) {
+            return true;
+        }
+        Set<String> roles =  new HashSet<>();
+        roles.add(ProjectConstants.OWNER);
+        roles.add(ProjectConstants.MEMBER);
+        Set<ProjectMemberRecord> projectMembers = new HashSet<>();
+        Condition allMembers = DSL.falseCondition();
+        for (ExternalId id: externalIds){
+            allMembers = allMembers.or(PROJECT_MEMBER.EXTERNAL_ID.eq(id.getId())
+                    .and(PROJECT_MEMBER.EXTERNAL_ID_TYPE.eq(id.getType()))
+                    .and(PROJECT_MEMBER.ROLE.in(roles))
+                    .and(PROJECT_MEMBER.PROJECT_ID.eq(projectId))
+                    .and(PROJECT_MEMBER.STATE.eq(CommonStatesConstants.ACTIVE))
+                    .and(PROJECT_MEMBER.REMOVED.isNull()));
+        }
+        projectMembers.addAll(create().selectFrom(PROJECT_MEMBER).where(allMembers).fetch());
+        return !projectMembers.isEmpty();
     }
 
-    @Inject
-    public void setObjectManager(ObjectManager objectManager) {
-        this.objectManager = objectManager;
+    @Override
+    public Account updateProject(Account project, String name, String description) {
+        Map<TableField<AccountRecord, String>, String> properties = new HashMap<>();
+        properties.put(ACCOUNT.NAME, name);
```

Comment:
name, description are optional parameters in the update. Should we add not null checks so when nothing is passed, the fields are not updated?